src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TrivialTemporaryStorageService.cs

```diff
@@ -4,29 +4,48 @@
 
 using System;
 using System.IO;
-using System.Text;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis;
 
 internal sealed class TrivialTemporaryStorageService : ITemporaryStorageServiceInternal
```

Comment:
it's very unclear if we need this type at all.  i believe the only use case is if we're on a platform that doesn't support MMFs *and* we're attempting to make skeleton references.  That's the only case i can see that would his this.  

The reference+oop case doesn't hit this as that's windows-only (because it's VS only) and thus we always have a real storage service there.

Reply:
and vscode should be on .net core, which i think we always have MMFs.  but i'm leaving this to a fiture investigation.